pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/impl/DockerPipelineFromDockerfileScript.groovy

```diff
@@ -39,12 +41,39 @@ public class DockerPipelineFromDockerfileScript extends DeclarativeAgentScript<D
 
     @Override
     public Closure run(Closure body) {
-        String targetLabel = describable.label
-        if (targetLabel == null) {
-            targetLabel = script.dockerLabel()?.trim()
+        if (describable.reuseNode && script.getContext(FilePath.class) != null) {
```

Comment:
... because even if `reuseNode` is always true this would only take effect if we already are on a node.